tuxemon/core/states/world/worldstate.py

```diff
@@ -502,13 +502,19 @@ def get_collision_map(self):
         # TODO: overlapping tiles/objects by returning a list
         collision_dict = dict()
 
+        # tile layout must be set first
+        collision_dict.update(self.collision_map)
+
         # Get all the NPCs' tile positions
         for npc in self.get_all_entities():
             pos = nearest(npc.tile_pos)
-            collision_dict[pos] = {"entity": npc}
 
-        # tile layout takes precedence
-        collision_dict.update(self.collision_map)
+            # make sure the data for this position exists
+            if not pos in collision_dict or not collision_dict[pos]:
+                collision_dict[pos] = {}
```

Comment:
Just in case the necessity of this check is questioned: Sometimes collision_dict doesn't contain the pos parameter. At other times it does but its value is None. In both cases it must be initialized as a dict, so that below we either add the entity to the existing parameters or this empty dict instead if none were found here.